### DISCUSSION TITLE
Shouldn't prompt formats be an enumerator?

### DISCUSSION THREAD

**Discussion by danemadsen** [ggml-org/llama.cpp#6335] [Q&A] [answered]
Hi i was looking into the code for gpt_params and I'm wondering why there's a boolean property for each prompt format (currently just `instruct` for Alpaca format and `chatml` for ChatML)? 

Shouldn't this just be a single enumerator property named something like `prompt_format` or something? 

```(params.instruct == true && params.chatml == true)``` should never be possible so it doesn't make sense to me why they each have a property.

**Comment by ggerganov** [comment, marked as answer]
These parameters are old "hacks" - should be removed and replaced with `llama_chat_apply_template()` in order to use the correct chat template that comes with the model
